### PR TITLE
gollm fixes and improvements

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* changes to the taskhanlder now that it is using a new event source library
* bugfix to gollm controller
* model config improvements

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ void init() {
 	})
 	public ResponseEntity<TaskResponse> createModelCardTask(
 			@RequestParam(name = "document-id", required = true) final UUID documentId,
-			@RequestParam(name = "mode", required = false, defaultValue = "async") final TaskMode mode) {
+			@RequestParam(name = "mode", required = false, defaultValue = "ASYNC") final TaskMode mode) {
```
Comment (kbirk) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java:86]: These _should_ be fixed to work with lowercase on the latest version of `main`. Sorry about that. apparently defining an enum as `TYPE("type")` in java does nothing.

Comment (blanchco) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java:86]: I just tried on the latest version of main with the lower cases and am still getting issues with the conversion:
`Failed to convert value of type 'java.lang.String' to required type 'software.uncharted.terarium.hmiserver.service.tasks.TaskService$TaskMode'; Failed to convert from type [java.lang.String] to type [@org.springframework.web.bind.annotation.RequestParam software.uncharted.terarium.hmiserver.service.tasks.TaskService$TaskMode] for value [async]`

**Review by YohannParis — APPROVED** [review]
Let's wait until this `ASYNC/async` parameter is resolve.

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java (hunk 1 above)
Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java:86]: Argh I guess my change didn't make it into `main`. Pushed it to this branch. Can you give it another try @blanchco

Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java:86]: Hold on that's another enum. lol. Java enums are ridiculous

Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java:86]: Okay Spring is stupid and doesn't deal with enums in any sane way. Leave these as upper case.

**Review by kbirk — APPROVED** [review]

**Comment by kbirk** [comment]
> Let's wait until this `ASYNC/async` parameter is resolve.

I was wrong, my update didn't fix the enum casings. Lets merge this we can fix the casing stuff another day

**Pull request merged**